### DISCUSSION TITLE
Rotor with polars as a function of Mach number

### DISCUSSION THREAD

**Discussion by eversilia** [byuflowlab/FLOWUnsteady#155] [Q&A] [answered]
Hi, 
can I define a blade with multiple airfoil polars that describe CL,CD and CM as a function of AoA **and Mach** ?
In the rotor example, the blade is defined without taking into account compressibility effects as far as I can tell !
Thank you in advance !

**Comment by cibinjoseph** [comment]
FLOWUnsteady has the provision to receive individual airfoil data tables for each section. You could provide separate Mach corrected airfoil data for each section.

**Comment by eversilia** [comment]
Thank you for your reply !
That is for sure duable for a propeller or in a rotor hovering condition.
On the other hand, in forward flight, the single section (depending on its position) can experience big variation in mach number. In this case we need the code to automatically query coefficients depending on the instantaneous Mach.

**Comment by cibinjoseph** [comment]
Ah, that's right. Should be possible to include that provision down the line as the code matures.

**Comment by EdoAlvarezR** [comment]
Related to #136.

As Cibin said, the capabilities of re-calculating (or providing) polars throughout the maneuver might be developed in the future.

**Comment by EdoAlvarezR** [comment, marked as answer]
@MarcoPic99 developed similar capabilities in #136, but for Reynolds number effects. I'd recommend doing the same for Mach number in simulations with variable RPM where compressibility effects might be important.
